pkgs/servers/dnschain/default.nix

```diff
@@ -1,11 +1,51 @@
-{ stdenv, nodePackages }:
-
-# to update dnschain, run npm2nix package.json package.nix, and
-# then add "coffee-script" manually as a dependecy for "dnschain"
-# in package.nix.
+{ stdenv, fetchFromGitHub, callPackage, makeWrapper, openssl }:
 
 let
-  np = nodePackages.override { generated = ./package.nix; self = np; };
-in
+  nodePackages = callPackage (import ../../top-level/node-packages.nix) {
```

Comment:
Yes, it produces the error:
```
called with unexpected argument ‘self’, at <nixpkgs>/lib/customisation.nix:56:12
```
I looked how other nodejs packages handle this and switched to use `callPackage`.